### DISCUSSION TITLE
Cant find the full open api spec for PrestaShop

### DISCUSSION THREAD

**Discussion by hackS3c101** [PrestaShop/PrestaShop#34391] [Q&A]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Is still an updated OpenAPI spec (aka Swagger) for API integration?
Could not find anywhere in the docs
Thanks 

### Steps to reproduce

Could not find anywhere

### PrestaShop version(s) where the bug happened

*

**Comment by florine2623** [comment]
ping @matks and @PrestaShop/prestashop-core-developers 
Are you aware of this option ?

**Comment by matks** [comment]
I converted this into a discussion as it's a question

@hackS3c101 for the current API which is called "Webservice" there is no such OpenAPI Spec, but you will find all the needed content here https://devdocs.prestashop-project.org/8/webservice/reference/ or, as it's mentioned, you can use the endpoint `?schema=blank` which will output a XML with all possible endpoints and format

**Comment by jolelievre** [comment]
I will add that we are implementing a new API based on APIPlatform and this one does include a Swagger doc, it is available on PrestaShop v9 only and you need to enable the `Authorization server` feature flag as it's still an experimental feature

You will then have access to the `Advanced Parameters > Authorization server` section in which you'll find a link to access the swagger doc (only accessible if you're logged into the BO) Also be aware that this is still a work in progress so very fe endpoints are available for now but we'll welcome contributions to add some (a documentation will be prepared once we finalize the initial MVP)
